tinker/faculty_bios/faculty_bio_controller.py

```diff
@@ -122,21 +124,160 @@ def _iterate_child_xml(self, child, author):
+                    'last-name': child.find('.//last').text or None,
+                    'deactivated': child.find('.//deactivate').text or None
+                }
+
+            # Returns the page_values
             return page_values
         else:
             return None
 
+    def csv_file(self, child):
+
+        # Initial dictionary values
+        page_values = {
+            'first': child.find('.//first').text,
+            'last': child.find('.//last').text,
+            'highlight': '',
+            'email': '',
+            'started-at-bethel': '',
+            'research-interests': '',
+            'areas': '',
+            'teaching-specialty': '',
+            'quote': '',
+            'website': ''
+        }
+
+        # Checks if there is data in each field, if not puts in an empty string
+        if child.find('.//highlight').text is not None:
+            page_values['highlight'] = child.find('.//highlight').text
+        if child.find('.//email').text is not None:
+            page_values['email'] = child.find('.//email').text
+        if child.find('.//started-at-bethel').text is not None:
+            page_values['started-at-bethel'] = child.find('.//started-at-bethel').text
+        if child.find('.//add-to-bio/research-interests').text is not None:
+            page_values['research-interests'] = child.find('.//add-to-bio/research-interests').text
+        if child.find('.//add-to-bio/areas').text is not None:
+            page_values['areas'] = child.find('.//add-to-bio/areas').text
+        if child.find('.//add-to-bio/teaching-specialty').text is not None:
+            page_values['teaching-specialty'] = child.find('.//add-to-bio/teaching-specialty').text
+        if child.find('.//add-to-bio/quote').text is not None:
+            page_values['quote'] = child.find('.//add-to-bio/quote').text
+        if child.find('.//add-to-bio/website').text is not None:
+            page_values['website'] = child.find('.//add-to-bio/website').text
+
+        # Iterates through all the job fields and adds it to the dictionary on the fly
+        max_jobs = 0
+        for jobs in child.iterfind('.//job-titles'):
+            max_jobs += 1
+            if str(jobs.find('school').text) == 'None':
+                school = ""
+            else:
+                school = jobs.find('school').text
+            if str(jobs.find('job_title').text) == 'None':
+                job_title = ""
+            else:
+                job_title = jobs.find('job_title').text
+            department = ""
+            page_values['school' + str(max_jobs)] = school
+            page_values['job_title' + str(max_jobs)] = job_title
+            page_values['department' + str(max_jobs)] = department
+            if school == 'Bethel University':
+                page_values['department' + str(max_jobs)] = department
+            elif school == 'College of Arts and Sciences':
+                if str(jobs.find('department').text) != 'None':
```

Comment:
these if's are also pretty repetitive. You could consider pulling the logic out to a method

Reply:
Used the same method I made for the comment above and pulled out some logic with it.

Reply:
👍